# Analysis of Covid Tweets to identify the usage of the word 'test' and its frequency

## Gather Data

In [4]:
# Import the libraries and dataset
import pandas as pd
import numpy as np

covid_df = pd.read_csv('covid19_tweets.csv')
covid_df.head()

user_name         user_location  \
0           ᏉᎥ☻լꂅϮ            astroworld   
1    Tom Basile 🇺🇸          New York, NY   
2  Time4fisticuffs      Pewee Valley, KY   
3      ethel mertz  Stuck in the Middle    
4         DIPR-J&K     Jammu and Kashmir   

                                    user_description     user_created  \
0  wednesday addams as a disney princess keepin i...   5/26/2017 5:46   
1  Husband, Father, Columnist & Commentator. Auth...  4/16/2009 20:06   
2  #Christian #Catholic #Conservative #Reagan #Re...  2/28/2009 18:57   
3  #Browns #Indians #ClevelandProud #[]_[] #Cavs ...    3/7/2019 1:45   
4  🖊️Official Twitter handle of Department of Inf...   2/12/2017 6:45   

   user_followers  user_friends  user_favourites  user_verified  \
0             624           950            18775          False   
1            2253          1677               24           True   
2            9275          9525             7254          False   
3             197           987             1488          False   
4          101009           168              101          False   

              date                                               text  \
0  7/25/2020 12:27  If I smelled the scent of hand sanitizers toda...   
1  7/25/2020 12:27  Hey @Yankees @YankeesPR and @MLB - wouldn't it...   
2  7/25/2020 12:27  @diane3443 @wdunlap @realDonaldTrump Trump nev...   
3  7/25/2020 12:27  @brookbanktv The one gift #COVID19 has give me...   
4  7/25/2020 12:27  25 July : Media Bulletin on Novel #CoronaVirus...   

                            hashtags               source  is_retweet  
0                                NaN   Twitter for iPhone       False  
1                                NaN  Twitter for Android       False  
2                        ['COVID19']  Twitter for Android       False  
3                        ['COVID19']   Twitter for iPhone       False  
4  ['CoronaVirusUpdates', 'COVID19']  Twitter for Android       False

In [5]:
covid_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 155101 entries, 0 to 155100
Data columns (total 13 columns):
user_name           155101 non-null object
user_location       123729 non-null object
user_description    146342 non-null object
user_created        155101 non-null object
user_followers      155101 non-null int64
user_friends        155101 non-null int64
user_favourites     155101 non-null int64
user_verified       155101 non-null bool
date                155101 non-null object
text                155101 non-null object
hashtags            110555 non-null object
source              155048 non-null object
is_retweet          155101 non-null bool
dtypes: bool(2), int64(3), object(8)
memory usage: 13.3+ MB


This dataset has 155101 rows and 13 columns. However not all column details are important. Only the 'text' field data is being analyzed over the given period of time. So, need to clean the data.

## Clean

In [12]:
# remove all the unnecessary data by dropping the irrelavant columns
df = covid_df.copy()
df.drop(['user_name','user_location','user_description','user_created','user_followers','user_friends','user_favourites','user_verified','hashtags','source','is_retweet'],axis = 1,inplace=True)
df.head()

date                                               text
0  7/25/2020 12:27  If I smelled the scent of hand sanitizers toda...
1  7/25/2020 12:27  Hey @Yankees @YankeesPR and @MLB - wouldn't it...
2  7/25/2020 12:27  @diane3443 @wdunlap @realDonaldTrump Trump nev...
3  7/25/2020 12:27  @brookbanktv The one gift #COVID19 has give me...
4  7/25/2020 12:27  25 July : Media Bulletin on Novel #CoronaVirus...

In [13]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 155101 entries, 0 to 155100
Data columns (total 2 columns):
date    155101 non-null object
text    155101 non-null object
dtypes: object(2)
memory usage: 2.4+ MB
